### PR TITLE
Added Functionality Storage Controller Attributes to Storage Data Source

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
##### SUMMARY

Added Functionality Storage Controller Attributes to Storage Data Source

Relates OR Closes #0000

##### ISSUE TYPE

- Docs Pull Request
- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: redfish/provider/data_source_redfish_storage.go [hunk 1]
```diff
@@ -142,9 +148,635 @@ func (g *StorageDatasource) readDatasourceRedfishStorage(d models.StorageDatasou
...
+func newCacheSummary(input redfish.CacheSummary) models.CacheSummary {
+	return models.CacheSummary{
+		TotalCacheSizeMiB: types.Int64Value(int64(input.TotalCacheSizeMiB)),
+	}
+}
+
+// StorageSchema is a function that returns the schema for Storage
+func StorageSchema() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"odata_context": schema.StringAttribute{
```
Comment (rounak-adhikary) [inline redfish/provider/data_source_redfish_storage.go:322]: We generally do not expose odata fields, like odata_context,  odata_type, or *_odata_count from our providers. Please confirm if these fields are expected here.

Comment (Krishnan-Priyanshu) [reply redfish/provider/data_source_redfish_storage.go]: removed all the odata fields

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**